core/arch/arm32/kernel/thread.c

```diff
@@ -633,6 +655,7 @@ void thread_init_primary(const struct thread_handlers *handlers)
 	 * checks verifies that the offsets used in assembly code matches
 	 * what's used in C code.
 	 */
+#ifdef ARM32
```

Comment:
Sure I've considered having it in a static inline function in thread.h close to the struct and the defines, I haven't tried it but if it works it could be an option.